runtime/src/main/java/org/corfudb/runtime/view/replication/IReplicationProtocol.java

```diff
@@ -63,12 +66,33 @@
      * just performs multiple reads (possible in parallel).
      *
      * @param layout                The layout to use for the readAll.
-     * @param globalAddresses       A set of addresses to read from.
+     * @param globalAddresses       A list of addresses to read from.
+     * @return                      A map of addresses to committed
+     *                              addresses, hole filling if necessary.
+     */
+    default @Nonnull
+            Map<Long, ILogData> readAll(Layout layout, List<Long> globalAddresses) {
+        return globalAddresses.parallelStream()
+                .map(a -> new AbstractMap.SimpleImmutableEntry<>(a, read(layout, a)))
+                .collect(Collectors.toMap(r -> r.getKey(), r -> r.getValue()));
```

Comment:
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Replace this lambda with a method reference. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS1612)
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Replace this lambda with a method reference. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS1612)